### PR TITLE
Fixes #1457, int val output in TOML format, adds test

### LINKED ISSUE #1457: [bug] int val output in TOML format as floating point
**- Do you want to request a *feature* or report a *bug*?**
bug

**- What is the current behavior?**
TOML file output always renders numbers as float. This is *by design* in the [tomlify package](https://github.com/jakwings/tomlify-j0.4#known-problems) and requires custom "output replacer" logic to fix.

**- If the current behavior is a bug, please provide the steps to reproduce.**
In config.yml, specify collection format as `toml` and a field as `number` with `valueType` set to `int`.

**- What is the expected behavior?**
`int` should *not* be rendered as a `float` with a decimal point `789.0`. Instead, it should be rendered `789`.

**- Please mention your versions where applicable.**
Netlify CMS version: 1.9.2
Browser version:
Latest Chrome beta

### PR DESCRIPTION
**Pull request by slathrop**
**- Summary**
Fixes #1457, int val output in TOML format, adds test

**- Test plan**
Test added and passing for int and float

**- Description for the changelog**
Fix int val output in TOML format file

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 73f8bfcdc4d17a2d1464d09cec02d401fbb073f0

https://deploy-preview-1458--cms-demo.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 73f8bfcdc4d17a2d1464d09cec02d401fbb073f0

https://deploy-preview-1458--netlify-cms-www.netlify.com

File: src/formats/__tests__/frontmatter.spec.js [hunk 1]
```diff
@@ -192,7 +192,7 @@ describe('Frontmatter', () => {
     );
   });
 
-  it('should stringify YAML with --- delimiters when it is explicitly set as the format without a custom delimiter',
```
Comment (tech4him1) [inline src/formats/__tests__/frontmatter.spec.js]: If this file doesn't have any actual code changes, can we just clean up the diff?

File: src/formats/toml.js [hunk 2]
```diff
@@ -4,13 +4,19 @@ import moment from 'moment';
 import AssetProxy from 'ValueObjects/AssetProxy';
 import { sortKeys } from './helpers';
 
+// IE polyfill for Number.isInteger
```
Comment (tech4him1) [inline src/formats/toml.js]: We don't actually support IE, so this can be removed. 😃

File: src/formats/toml.js [hunk 3]
```diff
@@ -4,13 +4,19 @@ import moment from 'moment';
...
+Number.isInteger = Number.isInteger || (value => typeof value === "number" && isFinite(value) && Math.floor(value) === value);
+
 const outputReplacer = (key, value) => {
   if (moment.isMoment(value)) {
     return value.format(value._f);
   }
   if (value instanceof AssetProxy) {
     return `${ value.path }`;
   }
+  if (Number.isInteger(value)) {
```
Comment (tech4him1) [inline src/formats/toml.js]: I wonder if we should be checking `isSafeInteger` instead, so we don't accidentally output an invalid number? Or is that only a concern for computations and it shouldn't be a problem with `toString()`? That was the original problem that @iology noted for `tomlify` anyway.

File: src/formats/toml.js (hunk 2 above)
Comment (slathrop) [reply src/formats/toml.js]: Docs refer to IE11 [here](https://www.netlifycms.org/docs/add-to-your-site/#add-the-netlify-identity-widget). Couldn't find any other references to browser support in docs though.

...Will remove this polyfill shortly though, as I can confirm that the CMS won't load at all in IE11 right now.

File: src/formats/toml.js (hunk 3 above)
Comment (slathrop) [reply src/formats/toml.js]: I think `isInteger` is sufficient... we already know we aren't dealing with a string representation of an `int`-like `number` that could overflow or lose precision. We're dealing with a straight-up `number` that JavaScript believes to be an `int`. And right, no calculations here. We're just telling `tomlify` not to put decimal places on the thing, which we must do by giving it a string representation of what we want rendered in the TOML file.

File: src/formats/toml.js (hunk 2 above)
Comment (slathrop) [reply src/formats/toml.js]: ...BTW, I did find the discussion in issues [here](https://github.com/netlify/netlify-cms/issues/674#issuecomment-338775684) RE: browser support, with a summary at the end. 😃

Comment (tech4him1) [reply src/formats/toml.js]: Haha, feel free to PR a removal for that if you want.

cc/ @verythorough

**Comment by tech4him1** [comment]
> I think `isInteger` is sufficient... we already know we aren't dealing with a string representation of an `int`-like `number` that could overflow or lose precision. We're dealing with a straight-up `number` that JavaScript believes to be an `int`. And right, no calculations here. We're just telling `tomlify` not to put decimal places on the thing, which we must do by giving it a string representation of what we want rendered in the TOML file.

That was my original thought as well. I just wasn't sure if it was relevant because obviously the creator of `tomlify` did?

**Comment by iology** [comment]
Hi, I'm the author of "tomlify". Tomlify, which is schema-less, does not distinguish between floats and integer, and it only throw an error when the number is not finite. So if you need exact integers, you should examine the key paths and check the numbers with Number.isSafeInteger and take care of overflow and underflow in other code.

**Comment by slathrop** [comment]
OK, so @tech4him1, I understand the points made RE: `isSafeInteger`, but really the CMS is just in the role of serializing to a TOML file here based on what is handed to it. The CMS's job isn't to "check the math" so-to-speak. Are you satisfied with this bug-fix PR as-is now?

**Comment by tech4him1** [comment]
Yes, thank you for your time answering my questions. 😄

**Comment by tech4him1** [comment]
@slathrop By the way, I'm sorry if I came across as argumentative -- I really appreciate you taking the time to submit this PR. I just wanted to make sure I fully understood it since there was potential for problems. Thanks!

**Comment by slathrop** [comment]
@tech4him1 Not at all, but very thoughtful of you to consider it anyway. Thanks.

**Pull request merged**